tests/spark_like_test.py

```diff
@@ -943,3 +954,175 @@ def test_left_join_overlapping_column(pyspark_constructor: Constructor) -> None:
         "c": [4.0, 6.0, None],
     }
     assert_equal_data(result, expected)
+
+
+# Copied from tests/expr_and_series/median_test.py
+def test_median(pyspark_constructor: Constructor) -> None:
+    data = {"a": [3, 8, 2, None], "b": [5, 5, None, 7], "z": [7.0, 8, 9, None]}
+    df = nw.from_native(pyspark_constructor(data))
+    result = df.select(
+        a=nw.col("a").median(), b=nw.col("b").median(), z=nw.col("z").median()
+    )
+    expected = {"a": [3.0], "b": [5.0], "z": [8.0]}
+    assert_equal_data(result, expected)
+
+
+# copied from tests/expr_and_series/clip_test.py
+def test_clip(pyspark_constructor: Constructor) -> None:
+    df = nw.from_native(pyspark_constructor({"a": [1, 2, 3, -4, 5]}))
+    result = df.select(
+        lower_only=nw.col("a").clip(lower_bound=3),
+        upper_only=nw.col("a").clip(upper_bound=4),
+        both=nw.col("a").clip(3, 4),
+    )
+    expected = {
+        "lower_only": [3, 3, 3, 3, 5],
+        "upper_only": [1, 2, 3, -4, 4],
+        "both": [3, 3, 3, 3, 4],
+    }
+    assert_equal_data(result, expected)
+
+
+# copied from tests/expr_and_series/is_between_test.py
+@pytest.mark.parametrize(
+    ("closed", "expected"),
+    [
+        ("left", [True, True, True, False]),
+        ("right", [False, True, True, True]),
+        ("both", [True, True, True, True]),
+        ("none", [False, True, True, False]),
+    ],
+)
+def test_is_between(
+    pyspark_constructor: Constructor, closed: str, expected: list[bool]
+) -> None:
+    data = {"a": [1, 4, 2, 5]}
+    df = nw.from_native(pyspark_constructor(data))
+    result = df.select(nw.col("a").is_between(1, 5, closed=closed))
+    expected_dict = {"a": expected}
+    assert_equal_data(result, expected_dict)
+
+
+# copied from tests/expr_and_series/is_duplicated_test.py
+def test_is_duplicated(pyspark_constructor: Constructor) -> None:
+    data = {"a": [1, 1, 2], "b": [1, 2, 3], "level_0": [0, 1, 2]}
+    df = nw.from_native(pyspark_constructor(data))
+    result = df.select(
+        a=nw.col("a").is_duplicated(),
+        b=nw.col("b").is_duplicated(),
+        level_0=nw.col("level_0"),
+    ).sort("level_0")
+    expected = {
+        "a": [True, True, False],
+        "b": [False, False, False],
+        "level_0": [0, 1, 2],
+    }
+    assert_equal_data(result, expected)
+
+
+def test_is_nan(pyspark_constructor: Constructor) -> None:
+    data = {"a": [1.0, float("nan"), 2.0, None, 3.0]}
+    df = nw.from_native(pyspark_constructor(data))
+    result = df.select(nan=nw.col("a").is_nan())
+    expected = {"nan": [False, False, False, False, False]}
```

Comment:
Shouldn't this result in:

```suggestion
    expected = {"nan": [False, True, False, False, False]}
```
?

Reply:
I had this question in mind and pointed earlier [here](https://github.com/narwhals-dev/narwhals/pull/1721#discussion_r1903234642). Its because `nan` is replaced with `NULL` here. I guess `nan` should be treated as itself. WDYS?

https://github.com/narwhals-dev/narwhals/blob/2aa436d1de98a2aa0b16d852467969de0b68754d/tests/spark_like_test.py#L42

Reply:
Apologies for missing that!

Let's try to re-create a nan after creation. Maybe something along the following:


```py
    data = {"a": [0, 1.0, None]}
    df = nw.from_native(pyspark_constructor(data))
    result = df.select(nan=(nw.col("a")/nw.col("a")).is_nan())
```
I would expect this to result in `[True, False, False]`

Reply:
Hi @FBruzzesi, I tried this way, but 
```python
df.select(nan=(nw.col("a")/nw.col("a")).is_nan())
```
would return `NoneType`.

Alternatively, I simply tried replacing some value with `nan`, but `SparkLikeNamespace` has no attribute `when`. Any suggestions?

Reply:
I don't understand the pyspark rules. The square root of -1 would generate a NaN, yet I am unable to make it work with narwhals:

```py
frame = session.createDataFrame(pd.DataFrame({"a": [0, 1, None]}).replace({float("nan"): None}))
nw.from_native(frame).select(b = nw.col("a")**0.5).to_native()
```

> InvalidIntoExprError: Expected an object which can be converted into an expression, got <class 'NoneType'>
>
> Hint:
> - if you were trying to select a column which does not have a string
>   column name, then you should explicitly use `nw.col`.
>   For example, `df.select(nw.col(0))` if you have a column named `0`.
> - if you were trying to create a new literal column, then you 
>   should explicitly use `nw.lit`.
>   For example, `df.select(nw.lit(0))` if you want to create a new
>   column with literal value `0`.

Would you be ok to keep `is_nan` method as a follow up? Although it seems quite straightforward, we would like to be able to test it properly

Reply:
Sure, for now I'll push the changes without `is_nan` and followup it later.